City-of-Helsinki/drupal-module-helfi-tpr
# UHF-2604: Ontology word details

Adds import for "ontology word details" and adds a new entity "OntologyWordDetails" for them. When these are imported, they are automatically published if the relevant Unit entity is also published.

For now, these are only used to get information high school search filtering.

**How to test:**
* Setup up Kasko development environment.
* Check out Kasko branch `UHF-2604-kasko-ontology_word_details`.
* Get the changes for this module: `composer require drupal/helfi_tpr:dev-UHF-2604-ontology_word_details`
* Run database updates: `drush updb`
* Import configs: `drush cim`
* Clear caches: `drush cr`
* Make sure you have imported tpr_units including Helsinki's upper secondary schools (you can use [this](https://www.hel.fi/helsinki/fi/kasvatus-ja-koulutus/lukiot/opiskelu/lukioiden-painotukset/) list to check what schools there are).
  * Publish those upper secondary school units (and maybe also e.g. Swedish translations)
* Make sure you have a landing page with High school search component (and maybe also a Swedish translation).
  * ...and those published schools are added to that search.
* Run: `drush migrate:import tpr_ontology_word_details`
* Now open the page containing the High school search and you should be able to filter the results using the "Select a special emphasis or study programme" and "Select a special educational mission" fields.
  * Using the list mentioned above, check that the filtering works correctly: correct emphasis and educational missions for correct schools.

Also check the relevant site specific PR: https://github.com/City-of-Helsinki/drupal-helfi-kasvatus-koulutus/pull/37
